### ISSUE TITLE
ONNX model input & output names are not preserved in TensorFlow & TensorFlow Lite models

### ISSUE THREAD

**Issue by jpowie01** [PINTO0309/onnx2tf#178]
### Issue Type

Others

### onnx2tf version number

1.6.0

### onnx version number

1.12.0

### tensorflow version number

2.10.0

### Download URL for ONNX

Is generated as part of the provided code.

### Parameter Replacement JSON

```json
{}
```

### Current behaviour

Models with multiple input & output tensors don't preserve their names coming from ONNX file. Instead, they fallback to default names like "inputs_0", "inputs_1" (for input) and "Identity", "Identity_1" (for output). I've tried to use signature definitions but that doesn't work either.

Here is an example reproduction of this problem:

```python
import torch
import onnxruntime
import numpy as np
import onnx2tf
import tensorflow as tf
from tensorflow.lite.python import interpreter as tflite_interpreter


class Model(torch.nn.Module):
    def forward(self, x, y):
        return {
            "add": x + y,
            "sub": x - y,
        }


# Let's double check what PyTorch gives us
model = Model()
pytorch_output = model.forward(10, 2)
print("[PyTorch] Model Predictions:", pytorch_output)

# First, export the above model to ONNX
torch.onnx.export(
    Model(),
    {"x": 10, "y": 2},
    "model.onnx",
    opset_version=16,
    input_names=["x", "y"],
    output_names=["add", "sub"],
)

# And check its output
session = onnxruntime.InferenceSession("model.onnx")
onnx_output = session.run(["add", "sub"], {"x": np.array(10), "y": np.array(2)})
print("[ONNX] Model Inputs:", [o.name for o in session.get_inputs()])
print("[ONNX] Model Outputs:", [o.name for o in session.get_outputs()])
print("[ONNX] Model Predictions:", onnx_output)

# Now, let's convert the ONNX model to TF
onnx2tf.convert(
    input_onnx_file_path="model.onnx",
    output_folder_path="model.tf",
    non_verbose=True,
)

# Let's check TensorFlow model
tf_model = tf.saved_model.load("model.tf")
tf_output = tf_model(
    tf.constant((10,), dtype=tf.int64),
    tf.constant((2,), dtype=tf.int64),
)
print("[TF] Model Inputs:", [node.op.name for node in tf_model.inputs])
print("[TF] Model Outputs:", [node.op.name for node in tf_model.outputs])
print("[TF] Model Predictions:", tf_output)

# And TFLite model
interpreter = tflite_interpreter.Interpreter(model_path="model.tf/model_float32.tflite")
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
interpreter.set_tensor(input_details[0]['index'], np.array((10,), dtype=np.int64))
interpreter.set_tensor(input_details[1]['index'], np.array((2,), dtype=np.int64))
interpreter.invoke()
add = interpreter.get_tensor(output_details[0]['index'])
sub = interpreter.get_tensor(output_details[1]['index'])
print("[TFLite] Input Details:", [(i["index"], i["name"]) for i in input_details])
print("[TFLite] Output Details:", [(o["index"], o["name"]) for o in output_details])
print("[TFLite] Model Predictions - Add:", add)
print("[TFLite] Model Predictions - Sub:", sub)
```

Here is the output of the above script:

```
[PyTorch] Model Predictions: {'add': 12, 'sub': 8}
[ONNX] Model Inputs: ['x', 'y']
[ONNX] Model Outputs: ['add', 'sub']
[ONNX] Model Predictions: [array(12, dtype=int64), array(8, dtype=int64)]
WARNING:absl:Please consider providing the trackable_obj argument in the from_concrete_functions. Providing without the trackable_obj argument is deprecated and it will use the deprecated conversion path.
[TF] Model Inputs: ['inputs_0', 'inputs_1']
[TF] Model Outputs: ['Identity', 'Identity_1']
[TF] Model Predictions: [<tf.Tensor: shape=(1,), dtype=int64, numpy=array([12])>, <tf.Tensor: shape=(1,), dtype=int64, numpy=array([8])>]
[TFLite] Input Details: [(0, 'inputs_0'), (1, 'inputs_1')]
[TFLite] Output Details: [(2, 'Identity'), (3, 'Identity_1')]
[TFLite] Model Predictions - Add: [12]
[TFLite] Model Predictions - Sub: [8]
```

ONNX model shows the input and output names correctly in Netron:

```bash
$ netron model.onnx
```

<img width="1141" alt="ONNX" src="https://user-images.githubusercontent.com/6514553/217618140-9f67d3a1-9d41-4a4c-9122-1e1cbf231a90.png">

While TensorFlow Lite model doesn't have correct names:

```bash
$ netron model.tf/model_float32.tflite
```

<img width="1139" alt="TF Lite" src="https://user-images.githubusercontent.com/6514553/217620213-0387782d-5838-4193-a98e-7880e28824d3.png">

### Expected behaviour

I would expect the above script to give me the same names across all formats:

```
[PyTorch] Model Predictions: {'add': 12, 'sub': 8}
[ONNX] Model Inputs: ['x', 'y']
[ONNX] Model Outputs: ['add', 'sub']
[ONNX] Model Predictions: [array(12, dtype=int64), array(8, dtype=int64)]
WARNING:absl:Please consider providing the trackable_obj argument in the from_concrete_functions. Providing without the trackable_obj argument is deprecated and it will use the deprecated conversion path.
[TF] Model Inputs: ['x', 'y']
[TF] Model Outputs: ['add', 'sub']
[TF] Model Predictions: [<tf.Tensor: shape=(1,), dtype=int64, numpy=array([12])>, <tf.Tensor: shape=(1,), dtype=int64, numpy=array([8])>]
[TFLite] Input Details: [(0, 'x'), (1, 'y')]
[TFLite] Output Details: [(2, 'add'), (3, 'sub')]
[TFLite] Model Predictions - Add: [12]
[TFLite] Model Predictions - Sub: [8]
```

Also, I would expect Netron to show me correct input & output names like in the case of ONNX:

<img width="1141" alt="ONNX" src="https://user-images.githubusercontent.com/6514553/217618140-9f67d3a1-9d41-4a4c-9122-1e1cbf231a90.png">

**Comment by PINTO0309** [comment]
I know before I started building the tool that this problem is caused by the TensorFlow specification, regardless of whether there are multiple inputs or a single input. Therefore, the following statement has been included in the README since its initial release.

"If you do not like the generated TFLite OP name, edit it using [tflite2json2tflite](https://github.com/PINTO0309/tflite2json2tflite)."

On my source code side I have the model transformed by setting the names of the input OP and output OP, but TensorFlow ignores it.

**Comment by PINTO0309** [comment]
- onnx
  ![image](https://user-images.githubusercontent.com/33194443/217705196-c4579591-a6d2-42e5-a916-83e1a43806ff.png)

- tflite
  ![image](https://user-images.githubusercontent.com/33194443/217705466-708382a4-56d5-40bc-ab91-34afd3d5766d.png)

**Comment by PINTO0309** [comment]
The output of the JSON file generated as a temporary file takes a very long time, so we do not recommend using it in models larger than 100 MB.
https://github.com/PINTO0309/onnx2tf/releases/tag/1.6.2

```
-coion, --copy_onnx_input_output_names_to_tflite
  Copy the input/output OP name of ONNX to the input/output OP name of tflite.
  Due to Tensorflow internal operating specifications,
  the input/output order of ONNX does not necessarily match
  the input/output order of tflite.
  Be sure to check that the input/output OP names in the generated
  tflite file have been converted as expected.
  Also, this option generates a huge JSON file as a temporary file for processing.
  Therefore, it is strongly discouraged to use it on large models of hundreds
  of megabytes or more.
```

**Comment by jpowie01** [comment]
Thanks for providing a fix for this issue! I've tried to use it and indeed, it works well for TFLite models. However, I'm also relying on TF models for inference and processing, so it doesn't solve all of the problems I found.

I spent some time yesterday poking around, trying to find a solution that could work for me and my use cases. As a result, I've opened a Pull Request that adjusts the output tensor names and provides coherent inference experience across all the model formats (ONNX, TF, TFLite):
https://github.com/PINTO0309/onnx2tf/pull/185

Let me know what you think about it!
